src/Formula/Types.hs

```diff
@@ -617,3 +652,27 @@ lengthBound nLiterals maxLen =
   --   | n == minLen = 2^n * literalLength
   --   | n == literalLength = 2^n + lengthBound (n-1) literalLength (minLen,maxLen)
   --   | otherwise = 2^n * literalLength + lengthBound (n-1) literalLength (minLen,maxLen)
+
+
+withPercentRange :: Formula a => PercentRangeMode -> a -> Bool
+withPercentRange mode form =
+    posLiteralsNumber <= max upperBound (if upperLiteralNumberBound == 0 then 0 else 1)
+        && posLiteralsNumber >= max (if lowerLiteralNumberBound == 0 then 0 else 1) lowerBound
+  where
+    (posLiteralsNumber, totalLiteralsNumber, (lowerLiteralNumberBound,upperLiteralNumberBound)) = case mode of
+      ByTruthValues bounds ->
+        let
+          tableEntries = getEntries (getTable form)
+          trueEntries = filter (== Just True) tableEntries
+        in (length trueEntries, length tableEntries, bounds)
+      ByPositiveLiterals bounds ->
+        let
+          allLiterals = duplicateLiterals form
+          posLiterals = filter isPositive allLiterals
+        in (length posLiterals, length allLiterals, bounds)
```

Comment:
Ich fürchte, dass ich nicht ganz verstehe, wie das hier funktioniert bzw. ob es korrekt ist.

Zum Beispiel wundert mich, dass die Variable `posLiteralsNumber` in dem einen Fall an `length trueEntries` gebunden wird. Das scheint mir unlogisch.